starter-code/routes/passportRouter.js

```diff
@@ -1,22 +1,77 @@
-const express        = require("express");
-const router         = express.Router();
+const express = require("express");
+const authRoutes = express.Router();
 // User model
-const User           = require("../models/user");
+const User = require("../models/user");
 // Bcrypt to encrypt passwords
-const bcrypt         = require("bcrypt");
-const bcryptSalt     = 10;
+const bcrypt = require("bcrypt");
+const bcryptSalt = 10;
 const ensureLogin = require("connect-ensure-login");
-const passport      = require("passport");
+const passport = require("passport");
 
-
-
-router.get("/private-page", ensureLogin.ensureLoggedIn(), (req, res) => {
+authRoutes.get("/private-page", ensureLogin.ensureLoggedIn(), (req, res) => {
   res.render("passport/private", { user: req.user });
 });
 
+authRoutes.get("/signup", (req, res, next) => {
+  res.render("passport/signup.ejs");
+});
+
+authRoutes.post("/signup", (req, res, next) => {
+  const username = req.body.username;
+  const password = req.body.password;
+
+  if (username === "" || password === "") {
+    res.render("views/passport/signup.ejs", {
+      message: "Indicate username and password"
+    });
+    return;
+  }
+
+  User.findOne({ username }, "username", (err, user) => {
+    if (user !== null) {
+      res.render("views/passport/signup.ejs", {
+        message: "The username already exists"
+      });
+      return;
+    }
 
+    const salt = bcrypt.genSaltSync(bcryptSalt);
+    const hashPass = bcrypt.hashSync(password, salt);
 
+    const newUser = new User({
+      username,
+      password: hashPass
+    });
 
+    newUser.save(err => {
+      if (err) {
+        res.render("views/passport/signup.ejs", {
+          message: "Something went wrong"
+        });
+      } else {
+        res.redirect("/");
+      }
+    });
+  });
+});
+
+authRoutes.get("/logout", (req, res) => {
+  req.logout();
+  res.redirect("/login");
+});
+
+authRoutes.get("/login", (req, res, next) => {
+  res.render("passport/login");
+});
 
+authRoutes.post(
+  "/login",
+  passport.authenticate("local", {
+    successRedirect: "/",
+    failureRedirect: "/login",
+    failureFlash: true,
```

Comment:
You are setting "failureFlash" to true but you are not passing any error message to the view (see previous comment).